File: js/BeginnersSelection/07_KagamiMochi.js
// https://atcoder.jp/contests/abs/tasks/abc085_b

Object.prototype.let = function(f) {
  return f(this.valueOf());
}

require('fs')
  .readFileSync('/dev/stdin', 'utf8')
  .trim()
  .split('\n')
  .slice(1)
  .let(x => [...new Set(x)])
  .length
  .let(console.log);

// https://atcoder.jp/contests/abs/submissions/21476014

File: jl/BeginnersSelection/07_KagamiMochi.jl
# https://atcoder.jp/contests/abs/tasks/abc085_b

(
  readline()
  |> x -> parse.(Int, x)
  |> n -> map(_ -> readline(), 1 : n)
  |> unique
  |> length
  |> println
)

# https://atcoder.jp/contests/abs/submissions/22589243
